flink/v1.17/flink/src/main/java/org/apache/iceberg/flink/sink/IcebergFilesCommitter.java

```diff
@@ -207,7 +217,7 @@ public void snapshotState(StateSnapshotContext context) throws Exception {
     long checkpointId = context.getCheckpointId();
     LOG.info(
         "Start to flush snapshot state to state backend, table: {}, checkpointId: {}",
-        table,
+        tableSupplier.get(),
```

Comment:
I think it would be better to get the `table` once per commit, and keep using that table during the given commit operation. Fetching the table every time could cause concurrency issues/surprises which are hard to track/reproduce.

Reply:
I updated the committer so the table reload happens after the commit.

Reply:
nit: Maybe initTable for consistencies shake